addon/lib/infinity-promise-array.js

```diff
@@ -0,0 +1,2 @@
+import Ember from 'ember';
+export default Ember.ArrayProxy.extend(Ember.PromiseProxyMixin);
```

Comment:
Wondering what the choice of `ArrayProxy` was for?

Reply:
I guess it could be ObjectProxy actually...?  Is that what you mean?

Reply:
yeah that could be one option.  Only b/c it looks like you are concerned with the `content` in the `willInsertElement` method.

Reply:
The main reason here for the wrapper however is due to needing "something to mixin `Ember.PromiseProxyMixin`.  Without it, the bound view doesn't update when the promise resolves, meaning you never actually see the InfinityModel's content.

However - I did think about mixing `Ember.PromiseProxyMixin` directly into our `InfinityModel` (which is already an ArrayProxy).

Do you think this would make more sense?  The reason I didn't is because I think it may mean a fundamental change to how we're thinking about the InfinityModel...

Reply:
IE - right now our route still handles all things to do with data fetching.  Our InfinityModel only handles building params and storing content.

If we mixed `Ember.PromiseProxyMixin` into it - we'd need to store the initial promise on the InfinityModel (not a bad idea) but at that point we need to ask: "should the InfinityModel have the store injected, and be in charge of actually "making requests"?

at that point we may not even need the route mixin...

Reply:
@hhff Hm.  At first glance, I like the idea of having the separation of concerns and responsibilities b/w the data and fetching that data.  

What about the `_loadingMore` variable?  Could that be used as the boolean that renders/doesn't render the `infinity-loader` component?

Reply:
I agree re: seperation of concerns.  Lets keep the mixin <> model pattern for now!

`_loadingMore` won't help us here unfortunately - as the issue is that the initial change in  `infinityModel.content` (not InfinityPromise) after it's rendered doesn't trigger a template re-render.  When wrapped in a PromiseProxy it's fine (PromiseProxy is the same as a DS.RecordsArray in practice)

I tried a bunch of other stuff like manually triggering a `propertyDidChange` on `[]` and `content` etc after the `pushObjects` call, but nothing worked.  PromiseProxy is the only thing that did.

Reply:
If it works w/ ObjectProxy though, we should do that...!

Reply:
Ah that makes sense.  The template needs a new reference to an array or through updating the content on the proxy for it to re-render.  I'm digging this implementation the more I look at it.

Reply:
I wonder if the `_loadingMore` variable can be swapped out with `isPending` property on the infinityModel now?

Reply:
not all infinityModel instances have `isPending` (in the case I explained about "stashing" infinityModels).  It's only avail on the "first load" directly from a model hook